org.eclipse.scanning.test/src/org/eclipse/scanning/test/event/queues/QueueBeanFactoryTest.java

```diff
@@ -0,0 +1,123 @@
+package org.eclipse.scanning.test.event.queues;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.fail;
+
+import java.util.Arrays;
+import java.util.List;
+
+import org.eclipse.scanning.api.event.queues.IQueueBeanFactory;
+import org.eclipse.scanning.api.event.queues.beans.PositionerAtom;
+import org.eclipse.scanning.api.event.queues.beans.QueueAtom;
+import org.eclipse.scanning.api.event.queues.beans.SubTaskAtom;
+import org.eclipse.scanning.api.event.queues.beans.TaskBean;
+import org.eclipse.scanning.api.event.queues.models.QueueModelException;
+import org.eclipse.scanning.api.event.queues.models.SubTaskAtomModel;
+import org.eclipse.scanning.api.event.queues.models.TaskBeanModel;
+import org.eclipse.scanning.event.queues.QueueBeanFactory;
+import org.junit.Before;
+import org.junit.Test;
+
+public class QueueBeanFactoryTest {
+	
+	private IQueueBeanFactory qbf;
+	
+	@Before
+	public void setUp() {
+		qbf = new QueueBeanFactory();
+	}
+	
+	@Test
+	public void testAddRemoveAtoms() throws QueueModelException {
+		//Add positioner to the queue atom register
+		String reference = "testAtom";
+		PositionerAtom positAtom = new PositionerAtom(reference, "dummy", 10);
+		positAtom.setName("Set dummy to 10");
+		qbf.registerAtom(positAtom);
+		
+		List<String> atomReg = qbf.getQueueAtomRegister();
+		assertEquals("Should only be one queue atoms registered in the factory", 1, atomReg.size());
+		assertEquals("No atom with the expected reference ("+reference+") registered!", reference, atomReg.get(0));
+		
+		//Get the atom and check it's config
+		PositionerAtom dum = (PositionerAtom)qbf.getQueueAtom(reference);
+		assertFalse("No atom with the expected reference registered", dum == null);
+		
+		
+		PositionerAtom detXAtom = new PositionerAtom("setDetX", "dummy", 225);
+		detXAtom.setName("Set detX to 10");
+		qbf.registerAtom(detXAtom);
+//TODO?		atomReg = qbf.getQueueAtomRegister();
+		assertEquals("Should be two queue atoms registered in the factory", 2, atomReg.size());
+		SubTaskAtomModel simpleSubTask = new SubTaskAtomModel("mvDum", "Move dummy & set detector X position", atomReg);
+		qbf.registerAtom(simpleSubTask);
+//TODO?		atomReg = qbf.getQueueAtomRegister();
+		assertEquals("Should be three queue atoms registered in the factory", 3, atomReg.size());
+		qbf.unregisterAtom("setDetX");
+		assertEquals("Should be two queue atoms registered in the factory", 2, atomReg.size());
+		try {
+			qbf.getQueueAtom("setDetX");
+			fail("Fetching unregistered atom did not throw an exception");
+		} catch (QueueModelException qme) {
+			//expected
+		}
+
+		qbf.unregisterAtom("mvDum");
+		assertEquals("Should be one queue atoms registered in the factory", 1, atomReg.size());
+		try {
+			qbf.getQueueAtom("mvDum");
+			fail("Fetching unregistered atom did not throw an exception");
+		} catch (QueueModelException qme) {
+			//expected

```

Comment:
I'm testing that an exception is thrown when the supplied string doesn't exist in the QueueBeanFactory registry. If the exception *is not* thrown, then something has gone wrong - hence the fail() in the try block. There's no point reporting that the exception was caught, unless you want a log overflowing with reports.